compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -642,7 +642,8 @@ pub fn compute_inherent_assoc_ty_args<'a, 'b, 'tcx>(
         );
     }
 
-    match selcx.infcx.at(&cause, param_env).eq(DefineOpaqueTypes::No, impl_ty, self_ty) {
+    // Inherent associated types are unstable, so we change their behaviour to match the new solver's
+    match selcx.infcx.at(&cause, param_env).eq(DefineOpaqueTypes::Yes, impl_ty, self_ty) {
```

Comment:
Yeah, I don't buy this justification, haha.

I don't think we should just change this just "[IATs] are unstable". We shouldn't change this until we change all the other `DefineOpaqueTypes::No` in the solver, which is dependent on us fixing bubble properly.

Can you revert this? Also, even if we wanted to flip this, there's no test associated with this.